File: VB/WpfReportDesigner_local/LocalReportService.vb
Imports Microsoft.VisualBasic
Imports System.Collections.Generic
Imports DevExpress.XtraReports.Service

Namespace WpfReportDesigner_local
	Friend Class LocalReportService
		Inherits ReportService
		Private Shared layouts As New Dictionary(Of String, Byte())()

		Protected Overrides Sub SaveReportLayout(ByVal reportName As String, ByVal layoutData() As Byte)
			layouts(reportName) = layoutData
		End Sub

		Protected Overrides Function LoadReportLayout(ByVal reportName As String) As Byte()
			Dim layout() As Byte
			Return If(layouts.TryGetValue(reportName, layout), layout, MyBase.LoadReportLayout(reportName))
		End Function
	End Class
End Namespace

File: CS/WpfReportDesigner_local/LocalReportService.cs
using System.Collections.Generic;
using DevExpress.XtraReports.Service;

namespace WpfReportDesigner_local {
    class LocalReportService : ReportService {
        static Dictionary<string, byte[]> layouts = new Dictionary<string, byte[]>();

        protected override void SaveReportLayout(string reportName, byte[] layoutData) {
            layouts[reportName] = layoutData;
        }

        protected override byte[] LoadReportLayout(string reportName) {
            byte[] layout;
            return layouts.TryGetValue(reportName, out layout) ? layout : base.LoadReportLayout(reportName);
        }
    }
}
